src/Microsoft.DotNet.ApiCompat/src/Microsoft.DotNet.ApiCompat.Core/Rules/Compat/AttributeDifference.cs

```diff
@@ -78,24 +78,27 @@ public override DifferenceType Diff(IDifferences differences, ITypeDefinition im
 
         public override DifferenceType Diff(IDifferences differences, ITypeDefinitionMember impl, ITypeDefinitionMember contract)
         {
-            var implMethod = impl as IMethodDefinition;
-            var contractMethod = contract as IMethodDefinition;
-            if (implMethod == null || contractMethod == null)
+            if (impl == null || contract == null)
                 return DifferenceType.Unknown;
 
-            bool changed = CheckAttributeDifferences(differences, implMethod, implMethod.Attributes, contractMethod.Attributes);
-
-            IParameterDefinition[] method1Params = implMethod.Parameters.ToArray();
-            IParameterDefinition[] method2Params = contractMethod.Parameters.ToArray();
-            for (int i = 0; i < implMethod.ParameterCount; i++)
-                changed |= CheckAttributeDifferences(differences, method1Params[i], method1Params[i].Attributes, method2Params[i].Attributes, member: implMethod);
+            bool changed = CheckAttributeDifferences(differences, impl, impl.Attributes, contract.Attributes);
 
-            if (implMethod.IsGeneric)
+            var implMethod = impl as IMethodDefinition;
+            var contractMethod = contract as IMethodDefinition;
+            if (implMethod != null && contractMethod != null)
             {
-                IGenericParameter[] method1GenParams = implMethod.GenericParameters.ToArray();
-                IGenericParameter[] method2GenParam = contractMethod.GenericParameters.ToArray();
-                for (int i = 0; i < implMethod.GenericParameterCount; i++)
-                    changed |= CheckAttributeDifferences(differences, method1GenParams[i], method1GenParams[i].Attributes, method2GenParam[i].Attributes, member: implMethod);
+                IParameterDefinition[] method1Params = implMethod.Parameters.ToArray();
+                IParameterDefinition[] method2Params = contractMethod.Parameters.ToArray();
+                for (int i = 0; i < implMethod.ParameterCount; i++)
+                    changed |= CheckAttributeDifferences(differences, method1Params[i], method1Params[i].Attributes, method2Params[i].Attributes, member: implMethod);
+
+                if (implMethod.IsGeneric)
```

Comment:
```suggestion
                if (implMethod.IsGeneric && contractMethod.IsGeneric)
```
, maybe?  So that there's not some sort of bad behavior if they disagree on that principle?  (Presumably something would have already reported that level of a mismatch)

Reply:
I believe these wouldn't line up as a comparison in that case.